### ISSUE TITLE
Remove the $ in the properties for Vue 3 Compatibility

### ISSUE THREAD

**Issue by shawnwildermuth** [vuelidate/vuelidate#701]
- Start Date: (2020-09-18)
- Reference Issues: 
- Implementation PR: 

# Summary
As Vue 3 shows warnings when mixing in with $ properties, we should move the properties to simple properties instead of prefixing with $. 

# Basic example
Currently, methods and properties are all prefixed with $ (to prevent collision in the Options API) for example:

``` Javascript
    async function onSave() {
      const state = await model.value.$validate();
      console.log(state);
    }
```
$validate() should just be validate().

Similarly the object model uses $:

``` HTML
      <input v-model="name.$model" :class="{ error: name.$error }" />
      <div v-if="name.required.$invalid" class="errorText">Name is required</div>
      <div v-if="name.min.$invalid" class="errorText">Name must be 5 characters or more</div>
```

# Motivation
To improve compatibility with the Composition API. 

# Detailed design
In practice I'd like to just expose 

``` JavaScript
  setup() {
    const message = reactive({
      name: "",
      emailAddress: "",
      message: "",
    });

    const fields = toRefs(message);

    const rules = {
      name: { required, min: minLength(5) },
      emailAddress: { required, email },
      message: { required, min: minLength(10), max: maxLength(250) },
    };

    const model = useVuelidate(rules, fields);

    async function onSave() {
      const state = await model.value.$validate();
      console.log(state);
    }

    return {
      onSave,
      ...model.value,
    };
  },
```

See how I'd like to just pass in the model with the spread operator to flatten the API. In addition, I'd love the examples not to call the passed in variable as $v for the same reasons.

# Drawbacks
May be compatibility/collision with Options API.

# Alternatives

# Adoption strategy

# Unresolved questions

**Comment by shentao** [comment]
Hey @shawnwildermuth! Yup, this is also something we noticed when we updated the Vue dep to rc-11. We are now discussing what to do with `$v`. However, the trick you employed:

```js
return {
      onSave,
      ...model.value,
    };
```

is definitely a neat thing to do which we might want to consider too (thus removing $ from other methods).
Will get back to you once we put some more thoughts on that. Thanks for reporting!
